index.html

```diff
@@ -2622,178 +2649,285 @@ <h3><dfn>Writing or pushing content</dfn></h3>
               </li>
-              <!-- converting each record -->
-            </ol>
-          </li> <!-- converting message -->
+            </ul>
+          </li>
+          <li>
+            Otherwise, [= exception/throw =] a {{TypeError}} and abort
+            these steps.
+          </li>
         </ol>
-      </section>
+      </div>
+    </section>
+
+    <section><h3>Validating external type</h3>
+      <p>
+        The [[NFC-RTD]] specifies that external types MUST contain
+        the domain name of the issuing organization, a a colon and a type
+        name that is at least one character long.
+      </p>
+      <p>
+        The [[NFC-RTD]] specifies to omit the URN prefix “`urn:nfc:ext:`”
+        in the binary encoding of external type records, therefore
+        applications SHOULD NOT specify that when creating
+        <a>external type records</a>.
+      </p>
+      <p>
+        When the domain name is not provided, by default Web NFC uses the
+        <a href="https://html.spec.whatwg.org/multipage/origin.html#concept-origin-effective-domain">
+        effective domain of the origin</a> as domain name in external type
+        names when <a href="#mapping-external-data-to-ndef">creating external type records</a>. Therefore, applications MAY specify only the
+        colon and the external type name.
+      </p>
+      <p class="note">
+        The [[NFC-RTD]] mandates that external type names are represented
+        with the URN prefix “`urn:nfc:ext:`”, e.g. when reading
+        <a>NDEF messages</a>. However, since <a>external type records</a> are
+        distinguished by having the <a>TNF FIELD</a> set to `0x04`, there is
+        no risk seen for type name clashing. Also, there are
+        <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
+        W3C TAG recommendations</a> to avoid using URNs in the Web.
+        Therefore, Web NFC does not use the URN prefix neither when reading
+        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
+        before the type name.
+      </p>
+      <p>
+        To <dfn>validate external type</dfn> given a |input:USVString| run
+        these steps:
+      </p>
+      <ol class=algorithm id="validate-external-type">
+        <li>
+          If |input:string| is not a {{USVString}} or it is empty, or its
+          length exceeds 255 bytes, or it contains white spaces,
+          return `false`.
+        </li>
+        <li>
+          If |input| does not start with colon ('`:`'), return |input|.
+          <p class="issue">
+            Validating domain name needs an algorithm.
+          </p>
+        </li>
+        <li>
+          If |input| starts with colon ('`:`'), let |domain:string| be the
+          <a data-cite="url#concept-url-serializer">serialization</a> of the
+          <a href="https://html.spec.whatwg.org/multipage/origin.html#concept-origin-effective-domain">effective domain of
```

Comment:
no easier way to quote? check https://respec.org/xref/

Reply:
`<a data-cite="html">effective domain</a> of the {{Document/origin}} `

Thanks!